src/v/security/gssapi_authenticator.cc

```diff
@@ -142,8 +143,9 @@ ss::future<result<bytes>> gssapi_authenticator::authenticate(bytes auth_bytes) {
           [this, auth_bytes]() { return ssfcap(auth_bytes); });
     }
     case state::ssfreq: {
-        co_return co_await _worker.submit(
-          [this, auth_bytes]() { return ssfreq(auth_bytes); });
+        co_return co_await _worker.submit([this, auth_bytes, rules = _rules]() {
+            return ssfreq(auth_bytes, rules);
+        });
```

Comment:
I'm specifically making a copy of the rules that's contained within the gssapi_authenticator in order to move it to the alien thread.